cmake/FindESMF.cmake

```diff
@@ -72,6 +72,19 @@ if (ESMF_FOUND)
     endif()
   endforeach()
 
+  # Construct ESMF_VERSION from ESMF_VERSION_STRING_GIT
+  if(ESMF_FOUND)
+    # ESMF_VERSION_MAJOR and ESMF_VERSION_MINOR are defined in ESMFMKFILE
+    set(ESMF_VERSION 0)
+    set(ESMF_VERSION_PATCH ${ESMF_VERSION_REVISION})
+        set(ESMF_BETA_RELEASE FALSE)
+    if(ESMF_VERSION_BETASNAPSHOT MATCHES "^('T')$")
+      set(ESMF_BETA_RELEASE TRUE)
+      message(STATUS "Beta release of ESMF detected.")
```

Comment:
It would still be good to echo what the beta snapshot it detected.
Perhaps something like:
```
string(REGEX REPLACE ".*beta_snapshot_*\([0-9]*\).*" "\\1" ESMF_BETA_SNAPSHOT "${ESMF_VERSION_STRING_GIT}")
message(STATUS "Detected ESMF Beta snapshot ${ESMF_BETA_SNAPSHOT}")
```